assets/src/edit-story/components/panels/stylePreset/index.js

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+import { rgba } from 'polished';
+import { useCallback, useEffect, useState } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { ReactComponent as Add } from '../../../icons/add_page.svg';
+import { ReactComponent as Edit } from '../../../icons/edit_pencil.svg';
+import { ReactComponent as Remove } from '../../../icons/remove.svg';
+import { useStory } from '../../../app/story';
+import generatePatternStyles from '../../../utils/generatePatternStyles';
+import { getDefinitionForType } from '../../../elements';
+import { Panel, PanelTitle, PanelContent } from './../panel';
+
+const COLOR_HEIGHT = 35;
+
+const buttonCSS = css`
+  background: transparent;
+  width: 30px;
+  height: 28px;
+  border: none;
+  color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.84)};
+  cursor: pointer;
+  padding: 0;
+`;
+
+// Since the whole wrapper title is already a button, can't use button directly here.
```

Comment:
Note sure how accessible that is 🤔 

For example, I cannot tab to those buttons right now because of this.

We should revisit this for beta.

Reply:
Oh, that's a good point, perhaps this panel could have its own non-button title which includes buttons inside -- it's customizing the default one quite a lot.

Will add a todo note for that.

Reply:
Tabbing is not required, if arrow keys work though. I'd actually prefer arrow navigation over tabbing to not pollute global tab order too much. Tabbing to the color presets should focus the first one, tabbing again should then move on to the next tab-able item after the presets.

When any preset is focused, arrow keys should move focus between the displayed presets. When tabbing away from presets and then back to the presets, the last focused preset should be focused again. This follows general WAI-ARIA recommendations for similar components.

Technically this can be implemented with each preset being a `role=button` with `tabIndex={activeIndex===index ? 0 : -1}`. And then use `useKeyDownEffect` to move focus - left-right simply increments/decrements `activeIndex` and up-down subtracts/adds `rowCount`.

Reply:
FWIW I was referring to tabbing to the "plus" and "edit" buttons within the panel `button`.

But your points regarding the preset buttons make sense to me.

Reply:
Hmm, actually, each color preset is already a button, it's the Edit and Add icons that aren't currently buttons and aren't tabbable (since the Title itself as the container is currently a button and doesn't allow extra button children).

The color presets are missing keydown handler at this point apparently but are tabbable.

Reply:
I now see this comment was misplaced. Please re-apply where appropriate.

Reply:
All the comments except for this are addressed now.

The color presets are currently buttons and they are usable with tabbing, kind of. Definitely arrow movements make more sense, however, I'd suggest leaving this part together with reworking the Title for Color Presets generally for the next version of the Color Presets (unless alpha gets pushed back another week). Thoughts?

Reply:
I'm okay with leaving arrow navigation for a follow-up ticket.